### PR TITLE
Refactor plugin structure

### PR DESCRIPTION
**Pull request by saberduck**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
     "clear": "tsc -b src tests --clean",
     "check-format": "prettier --list-different \"{src,tests}/**/!(*.lint).ts\"",
     "test": "jest",
+    "test-with-coverage": "jest --coverage",
     "ctest": "jest tests/tools/testers/comment-based/launcher.test.ts --verbose=false",
     "format": "prettier --write \"{src,tests,tools,profiling}/**/!(*.lint).ts\"",
     "compile": "tsc -b src tests profiling",
-    "jar": "npm pack && mkdirp target/classes && mv eslint-bridge-1.0.0.tgz target/classes && bash tools/check-distribution-filepath-length.sh",
+    "check-distribution": "bash tools/check-distribution-filepath-length.sh",
+    "mvn-copy": "cd sonar-plugin && mkdirp sonar-javascript-plugin/target/classes && cp ../sonarjs-1.0.0.tgz sonar-javascript-plugin/target/classes",
```
Comment (francoismora) [inline package.json:15]: SonarLint VisualStudio has the filename `eslint-bridge-1.0.0.tgz` [defined](https://github.com/SonarSource/sonarlint-visualstudio/blob/63a9400ef82f8e62298b8ab2eb433723dc17a8f8/build/DownloadSonarPlugin/DownloadAndExtractSonarJS.cs#L54) on their side too. So they should change it, too, after the PR.

**Review by francoismora — APPROVED** [review]
LGTM

**Review by ilia-kebets-sonarsource — CHANGES_REQUESTED** [review]
- The `npm new-rule` script has broken paths
- The paths in `docs/DEV.md` still reference `eslint-bridge/`

I've opened a [PR](https://github.com/SonarSource/SonarJS/pull/3733#event-8616183500) with changes where I fix stuff for typedoc.

Otherwise, LGTM, I like the fresh makeover!

File: package.json (hunk 1 above)
Comment (saberduck) [reply package.json:15]: yes

**Review by ilia-kebets-sonarsource — APPROVED** [review]

**Pull request merged**